File: codes/data_analysis/testing_labs_map.R
library(sf)

# Load the Netherlands shapefile and set CRS
netherlands <- st_read(
    "./data/source/polygons/wijkenbuurten_2024_v1.gpkg",
    layer = "buurten"
)
netherlands <- st_transform(netherlands, crs = 4326)

# Filter out water areas
netherlands_nowater <- netherlands |> filter(water != "JA")

# Load the testing facilities data and convert to sf object
test_fac <- read.csv(
    "./data/source/testing_facilities/drug_testing_facilities.csv"
)

test_fac <- st_as_sf(
    test_fac,
    coords = c("longitude", "latitude"),
    crs = 4326
)

# Plot the data
dutch_map <- ggplot() +
    geom_sf(
        data = netherlands_nowater,
        fill = "#538a95",
        color = "white",
        size = 0.05
    ) +
    geom_sf(
        data = test_fac,
        color = "#565175",
        size = 5
    ) +
    theme_void()

ggsave(
    filename = "results/figures/descriptives/drug_testing_facilities_map.png",
    plot = dutch_map,
    dpi = 600,
    height = 8,
    width = 8
)

File: codes/data_analysis/testing_labs_map.py
import pandas as pd
import geopandas as gpd
import matplotlib.pyplot as plt

netherlands = gpd.read_file(r'./data/source/polygons/wijkenbuurten_2024_v1.gpkg', layer='buurten')
netherlands = netherlands.to_crs(4326)
netherlands_nowater = netherlands[netherlands['water'] != 'JA']

test_fac = pd.read_csv(r'./data/source/testing_facilities/drug_testing_facilities.csv')
test_fac = gpd.GeoDataFrame(
    test_fac,
    geometry = gpd.points_from_xy(
        x = test_fac.longitude,
        y = test_fac.latitude,
        crs = 'EPSG:4326',
    )
)

fig, ax = plt.subplots()
netherlands_nowater.plot(ax=ax, color='#538a95', edgecolor='white', linewidth=0.15)
test_fac.plot(ax=ax, color='#565175', markersize=20)

plt.axis('off')
plt.savefig(r'./results/figures/descriptives/drug_testing_facilities_map.png', dpi=600, bbox_inches='tight')